heat/core/dndarray.py

```diff
@@ -887,6 +881,12 @@ def __getitem__(self, key: Union[int, Tuple[int, ...], List[int, ...]]) -> DNDar
             balanced=True if new_split is None else None,
         )
 
+    @staticmethod
+    def __getitem_is_key_singular(key: any, axis: int, self_proxy: torch.Tensor) -> bool:
```

Comment:
I agree with abstracting this method, I'm confused by the name. Something like `__key_is_singular` would work well I think, I'm sure we'll end up using this in `setitem` as well.

Reply:
okay that works for me. ill make the change now. my motivation behind the name was to keep it right below the getitem function

Reply:
ive changed it to `__is_key_singular` to keep it nearby